diff --git a/src/GPU3D_Compute.cpp b/src/GPU3D_Compute.cpp
index 0000000..0000000 100644
--- a/src/GPU3D_Compute.cpp
+++ b/src/GPU3D_Compute.cpp
@@ -1033,7 +1033,10 @@ void ComputeRenderer::RenderFrame(GPU& gpu)
                 }
 
                 glUniform1ui(UniformIdxCurVariant, i);
-                glUniform2f(UniformIdxTextureSize, 1.f / variants[i].Width, 1.f / variants[i].Height);
+                if (variants[i].Texture != 0)
+                    glUniform2f(UniformIdxTextureSize,
+                        1.f / variants[i].Width,
+                        1.f / variants[i].Height);
                 glBindBuffer(GL_DISPATCH_INDIRECT_BUFFER, BinResultMemory);
                 glDispatchComputeIndirect(offsetof(BinResultHeader, VariantWorkCount) + i*4*4);
             }
